cylc/flow/workflow_files.py

```diff
@@ -1616,20 +1609,19 @@ def check_flow_file(
             return flow_file_path
         if flow_file_path.resolve() == suite_rc_path.resolve():
             # A symlink that points to *existing* suite.rc
-            if logger:
-                logger.warning(SUITERC_DEPR_MSG)
             return flow_file_path
     if suite_rc_path.is_file():
         if not symlink_suiterc:
-            if logger:
-                logger.warning(SUITERC_DEPR_MSG)
```

Comment:
I'm not sure these warnings should be removed. Currently there is no warning when doing `cylc play` on a `suite.rc` workflow

Reply:
I think I've got it working now, for validate and play, with a single log call in the central reg parse function.